build.sbt

```diff
@@ -6,14 +6,13 @@ scalaVersion := "2.11.7"
 
 libraryDependencies += "org.spire-math" %% "cats" % "0.2.0"
 libraryDependencies += "com.lihaoyi" %% "pprint" % "0.3.6"
+libraryDependencies += "com.lihaoyi" %% "upickle" % "0.3.6"
 libraryDependencies += "com.chuusai" %% "shapeless" % "2.2.5"
 
 resolvers += Resolver.sonatypeRepo("releases")
 
 addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.6.3")
 
-wartremoverErrors ++= Warts.allBut(Wart.Any, Wart.Nothing, Wart.Throw, Wart.IsInstanceOf, Wart.AsInstanceOf, Wart.NoNeedForMonad)
```

Comment:
I like wartremover.. when it works, but it seems to restrict libraries we can use conveniently at this point.. e.g. uPickle fails quite dramatically and I couldnt find a way to selectively disable it (as it fails when looking for implicits).. 

do you have any ideas how can I restore it? (should I use another serialisation library?)